server/dive_tasks/tasks.py

```diff
@@ -611,46 +623,23 @@ def extract_zip(self: Task, folderId: str, itemId: str):
             f"/item/{str(item['_id'])}?folderId={str(created_folder['_id'])}",
         )
 
-        if len(top_level_folders) > 0:
-            for top_folder in top_level_folders.keys():
-                base_folder = ''  # default for no subfolder
-                # single flat/dataset import
-                if len(top_level_folders) != 1:
-                    base_folder = top_folder
-                # multiple flat/dataset import
-                if top_level_folders[top_folder] == 'flat':
-                    utils.upload_zipped_flat_media_files(
-                        gc,
-                        manager,
-                        folderId,
-                        Path(f"{_working_directory}/{top_folder}"),
-                        base_folder,
-                    )
-                # multiple flat file exports combined
-                elif top_level_folders[top_folder] == 'dataset':
-                    utils.upload_exported_zipped_dataset(
-                        gc,
-                        manager,
-                        folderId,
-                        Path(f"{_working_directory}/{top_folder}"),
-                        base_folder,
-                    )
-
-        else:  # flat zip file with media and/or annotation data inside
-            utils.upload_zipped_flat_media_files(gc, manager, folderId, Path(_working_directory))
-
-        # remove auxilary folder and metadata if there are sub folders
-        if len(top_level_folders) > 1:
-            sub_folders = list(gc.listFolder(folderId))
-            aux_folders = list(
-                (x for x in sub_folders if x['name'] == constants.AuxiliaryFolderName)
-            )
-            if len(aux_folders) > 0:
-                gc.delete(f"folder/{aux_folders[0]['_id']}")
-            # remove metadata
-            metdata = [constants.TypeMarker, constants.FPSMarker, constants.DatasetMarker]
-            gc.sendRestRequest(
-                "DELETE",
-                f"folder/{folderId}/metadata",
-                json=metdata,
-            )
+        for folderName, folderType in discovered_folders.items():
+            subFolderName = folderName if len(discovered_folders) > 1 else ''
```

Comment:
The removal of the check for a single top level folder means that it imports anything from the exported Zip from the website as a subFolder instead of directly importing it in. I.E. - if you export a dataset as zip using the export all and then try importing it. Instead of treating it as the ./Import_name, it will treat it as ./Import_name/dataset_name meaning that it will always import the exported zips as a sub folder of the import. For multiple sub folders this makes sense but I'm not sure if we want this behavior for a single export zip dataset.  I realize this update now allows arbitrary depth of the imports, but at the cost of a smooth export/import procedure for the zip exports.  Just want to bring that up and see if it's worth it.

The idea would be to do a sum of the number of 'unstructured' and 'datatset' folders and if it is 1 the `subFolderName = ''` else it sets it to the current subfolder name?

Reply:
Done.